### PR TITLE
warn users before attempting to use app without passcode

### LINKED ISSUE #338: If device pin/biometrics not set, provide instructions to do so
With changes from issue #314, lock capabilities  are unusable in the settings screen when users do not have any available device authentication methods.

We need to communicate and support users to set this up without requiring device authentication to use the app.

## Requirements

- display the passcode prompt (when appropriate) on `Get Started` button tap in the welcome page
- if the user taps `Skip`, just route to FxA
- if the user hits set passcode, route to the settings page and remain on the welcome screen in the application
- don't bother storing the `Skip` preference, just launch it (when warranted) every time the user hits `Get Started`

## Visual Design
**InVision**
https://mozilla.invisionapp.com/share/7JEUG47ZU#/296414673_No_PIN

**Zeplin**
https://zpl.io/V4JW1Jz

Note: "Set Passcode" button would direct user to Touch ID & Passcode section of Settings app.

See also #339

### PR DESCRIPTION
**Pull request by sashei**
resolves #338

### CODE REVIEW

**Comment by devinreams** [comment]
I may be missing something but I don't believe there's a way to simulate passcode presence or not (just biometrics) on Simulator and removing a passcode from a live iPhone results in a lot of hassle and mess: so we'Il especially need extra help from folks with willing 'test' devices on.

**Comment by sashei** [comment]
yeah I could not find a good way to test this in the simulator... physical device required.

File: lockbox-ios/Action/ExternalLinkAction.swift [hunk 1]
```diff
@@ -6,18 +6,30 @@ import Foundation
 import RxSwift
 import UIKit
 
-struct ExternalLinkAction: Action {
+protocol LinkAction: Action {}
+
+struct ExternalLinkAction: LinkAction {
     let url: String
```
Comment (jhugman) [inline lockbox-ios/Action/ExternalLinkAction.swift]: Nit: use `let url: URL`.

File: lockbox-ios/Action/ExternalLinkAction.swift [hunk 2]
```diff
@@ -32,8 +44,12 @@ class ExternalLinkActionHandler: ActionHandler {
...
-        self.openUrl(string: action.url)
+    func invoke(_ action: LinkAction) {
+        if let externalLink = action as? ExternalLinkAction {
+            self.openUrl(string: externalLink.url)
+        } else if let settingLink = action as? SettingLinkAction {
+            self.openSettings(settingLink)
+        }
     }
 
     private func openUrl(string url: String) {
```
Comment (jhugman) [inline lockbox-ios/Action/ExternalLinkAction.swift]: Nit: use `open(url: URL)`

**Pull request merged**